gg_fit_curve.py

```diff
@@ -3,75 +3,39 @@
 as it existed 2/24/2020.
 """
 from math import sqrt
+from picosvg.geometric_types import Point, Vector
 from typing import NamedTuple, Sequence, Tuple
 
 
-class Point(NamedTuple):
-    x: float
-    y: float
-
-    def magnitude(self) -> float:
-        return sqrt(self.x * self.x + self.y * self.y)
-
-    # V2AddII; it's weird this works on vectors...
-    def add(self, other) -> "Point":
-        return Point(self.x + other.x, self.y + other.y)
-
-    # V2SubII
-    def sub(self, other) -> "Point":
-        return Point(self.x - other.x, self.y - other.y)
-
-    def scale(self, s: float) -> "Point":
-        return Point(self.x * s, self.y * s)
-
-    @staticmethod
-    def tan_left(points: Sequence["Point"]) -> "Vector":
-        assert len(points) > 1
-        return Vector.p0_to_p1(points[0], points[1]).normalize()
-
-    @staticmethod
-    def tan_right(points: Sequence["Point"]) -> "Vector":
-        assert len(points) > 1
-        return Vector.p0_to_p1(points[-1], points[-2]).normalize()
-
-    @staticmethod
-    def tan_center(points: Sequence["Point"]) -> "Vector":
-        assert len(points) == 3
-        v1 = Vector.p0_to_p1(points[1], points[0])
-        v2 = Vector.p0_to_p1(points[2], points[1])
-        tHatCenter = Vector((v1.x + v2.x) / 2, (v1.y + v2.y) / 2)
-        return tHatCenter.normalize()
+CubicBezier = Tuple[Point, Point, Point, Point]
 
 
-CubicBezier = Tuple[Point, Point, Point, Point]
+def _unit(v: Vector) -> Vector:
+    v = v.unit()
+    if v is None:
+        v = Vector()
+    return v
 
 
-class Vector(Point):
-    def squared_length(self) -> float:
-        return self.x * self.x + self.y * self.y
+def _squared_length(v: Vector) -> float:
+    return v.x * v.x + v.y * v.y
 
-    def length(self) -> float:
-        return sqrt(self.squared_length())
 
-    def normalize(self) -> "Vector":
-        length = self.length()
-        x, y = 0.0, 0.0
-        if length != 0.0:
-            x = self.x / length
-            y = self.y / length
+def _tan_left(points: Sequence[Point]) -> Vector:
+    assert len(points) > 1
+    return _unit(points[1] - points[0])
 
-        return Vector(x, y)
 
-    def negate(self) -> "Vector":
-        return Vector(-self.x, -self.y)
+def _tan_right(points: Sequence[Point]) -> Vector:
+    assert len(points) > 1
+    return _unit(points[-2] - points[-1])
 
-    @staticmethod
-    def p0_to_p1(p0: Point, p1: Point) -> "Vector":
-        return Vector(*(p1.sub(p0)))
 
-    @staticmethod
-    def dot_product(v0: "Vector", v1: "Vector") -> float:
-        return v0.x * v1.x + v0.y * v1.y
+def _tan_center(points: Sequence[Point]) -> Vector:
+    assert len(points) == 3
+    v1 = points[0] - points[1]
+    v2 = points[1] - points[2]
+    return _unit(Vector((v1.x + v2.x) / 2, (v1.y + v2.y) / 2))
```

Comment:
instead of `Vector((v1.x + v2.x) / 2, (v1.y + v2.y) / 2)` you could do `(v1 + v2) * 0.5`?

Reply:
Given I'm taking the unit vector of the result I'm now unclear why divide by two at all